components/sysutils/scfdot/files/scfdotctl

```diff
@@ -0,0 +1,176 @@
+# page.  Otherwise, it won't print more than about 50" long.
+#
+
+# Not sure if this is exactly the right size, or if it will always precede
+# BeginSetup, but it works for dot-produced .ps files.
+/^%%BoundingBox:/ { width = \$4; height = \$5 }
+
+/^%%BeginSetup\$/ {
+	if (width == 0 || height == 0) {
+		exit 1;
+	}
+	print;
+
+	# I doubt this is necessary, but the PPD spec suggests it.
+	print "%%BeginFeature: *CustomPageSize";
+
+	# Here we print the parameters to the code below.  We add an inch of
+	# 'far' margin (the 'near' margin should have already been set by dot
+	# to be whatever margin was set to in the .dot file).
+	printf "%d %d\n", width + 72, height + 72;
+
+	# This is the magic code from HP's ppd file.  Actually, the ppd code
+	# begins with pop pop pop, but that just discards the orientation and
+	# offset parameters, so if we don't print them in the first place, it
+	# won't matter.
+	print "<</PageSize [ 5 -2 roll ] /ImagingBBox null>>setpagedevice";
+
+	print "%%EndFeature";
+	next;
+}
+
+{ print }
+EOF
+
+cat << EOF > /tmp/enlarge.awk.$$
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at CDDL.LICENSE.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at CDDL.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright 2005 Sun Microsystems, Inc.  All rights reserved.
+# Use is subject to license terms.
+#
+
+#
+# Enlarge the bounding box of a PostScript file by amounts specified on the
+# command line.  They should be specified by assignments to the left, right,
+# top, and bottom variables, in inches.  For example,
+#
+#	awk -f enlarge.awk file.ps top=2 bottom=2 > newfile.ps
+#
+
+BEGIN {
+	left = 0;
+	bottom = 0;
+	right = 0;
+	top = 0;
+}
+
+/^%%BoundingBox:/ {
+	printf "%s %d %d %d %d\n", \$1, \$2 - left * 72, \$3 - bottom * 72, \\
+	    \$4 + right * 72, \$5 + top * 72;
+	next;
+}
+
+{ print }
+EOF
+
+
+scfdot "${SCFDOTOPTSL}"      > /tmp/legend.dot.$$
+dot -Teps /tmp/legend.dot.$$ > /tmp/legend.ps.$$
+# awk doesn't like quoted argument 
+awk -f /tmp/enlarge.awk.$$ top=3 bottom=3 \
+      /tmp/legend.ps.$$      > legend.ps
+scfdot "${SCFDOTOPTSH}"      > /tmp/hostname.dot.$$
+echo "If you want to speed up the process use lower values for option Gmclimit (used value: ${DOTOPTS})"
+dot -Tps2 "${DOTOPTS}" < /tmp/hostname.dot.$$ > /tmp/hostname.ps.$$
+awk -f /tmp/setpage.awk.$$ /tmp/hostname.ps.$$ > hostname.ps
```

Comment:
Again, do we need to create hostname.ps and hostname.pdf in $(pwd)

Reply:
They are created in ${pwd} and then renamed to the final names ${HOSTNAME}.pdf respective ${HOSTNAME}.ps.

Reply:
can we store only final files in $(pwd) ?

Reply:
no problem, I will change it